### PR TITLE
Wallet authentication (move password to settings)

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

File: src/pages/mobile/SettingsPasswordSet.vue [hunk 1]
```diff
@@ -32,23 +20,34 @@
...
+        v-if="isWalletEncrypted"
         v-model="password"
         v-validate="'required|min:4'"
-        autofocus
-        autocomplete="new-password"
+        :autofocus="isWalletEncrypted"
+        autocomplete="current-password"
         :error="errors.has('password')"
         :footer="errors.first('password')"
         name="password"
```
Comment (CedrikNikita) [inline src/pages/mobile/SettingsPasswordSet.vue]:
```suggestion
         name="password"
         @input="wrongPassword = false"
```

File: src/pages/mobile/SettingsPasswordSet.vue [hunk 2]
```diff
@@ -32,23 +20,34 @@
...
+      />
```
Comment (CedrikNikita) [inline src/pages/mobile/SettingsPasswordSet.vue]:
```suggestion
      >
        <template slot="footer">
          <template v-if="errors.has('password')">
            {{ errors.first('password') }}
          </template>
          <template v-else-if="wrongPassword">
            Wrong password, try again
          </template>
        </template>
      </AeInputPassword>
```
In general i propose to show user if there is a wrong password entered.

File: src/components/mobile/PasswordModal.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <form
+      :id="_uid"
+      @submit.prevent="continueHandler"
+    >
+      <PasswordPurpose />
+      <AeInputPassword
+        v-model="password"
+        v-validate="'required|min:4'"
+        autofocus
+        autocomplete="current-password"
```
Comment (CedrikNikita) [inline src/components/mobile/PasswordModal.vue]: If our goal is to check that user remembers the password, should we autocomplete it?

File: src/pages/mobile/SettingsPasswordSet.vue [hunk 4]
```diff
@@ -32,23 +20,34 @@
     >
       <PasswordPurpose />
       <AeInputPassword
+        v-if="isWalletEncrypted"
         v-model="password"
         v-validate="'required|min:4'"
-        autofocus
-        autocomplete="new-password"
+        :autofocus="isWalletEncrypted"
```
Comment (CedrikNikita) [inline src/pages/mobile/SettingsPasswordSet.vue]: Same question about autocompletion in this case.

File: src/components/mobile/ConnectionStatus.vue [hunk 5]
```diff
@@ -16,6 +16,7 @@ export default {
     message: ({ onLine, sdk }) => {
       if (!onLine) return { text: 'You are offline... Please check your connection.' };
       if (!sdk) return { text: 'We are unable to connect to the chosen node.' };
+      if (sdk.then) return { text: 'Connecting to network...', className: 'connecting' };
```
Comment (CedrikNikita) [inline src/components/mobile/ConnectionStatus.vue]:
```suggestion
      if (sdk.then) return { text: 'Connecting to the network...', className: 'connecting' };
```
Or `to a network`

**Comment by CedrikNikita** [comment]
closes #429, closes #836, closes #837

File: src/components/mobile/PasswordModal.vue (hunk 3 above)
Comment (davidyuk) [reply src/components/mobile/PasswordModal.vue]: This is a hint for a web browser. Maybe the user has some software/hardware that handles user's passwords in some secure way. I don't think that we should be against it.

**Pull request merged**